src/shadertools/lib/shader-modules.js

```diff
@@ -84,28 +84,26 @@ export function getShaderDependencies(modules) {
   return Object.keys(result).sort((a, b) => result[b] - result[a]);
 }
 
-// PRIVATE API
-
-function registerShaderModule(shaderModule, {ignoreMultipleRegistrations = false}) {
-  assert(shaderModule.name, 'shader module has no name');
-  if (!ignoreMultipleRegistrations && shaderModules[shaderModule.name]) {
-    // TODO - instead verify that definition is not changing...
-    throw new Error(`shader module ${shaderModule.name} already registered`);
-  }
-  shaderModules[shaderModule.name] = shaderModule;
-  shaderModule.dependencies = shaderModule.dependencies || [];
-}
-
+/**
+ * Recursively checks module dpendencies to calculate dependency
+ * level of each module.
+ *
+ * @param {String[]} modules - Array of modules
+ * @param {Number} level - Current level
+ * @return {result} - Map of module name to its level
+ */
 // Adds another level of dependencies to the result map
-function getDependencyGraph({modules, level, result}) {
+export function getDependencyGraph({modules, level, result}) {
```

Comment:
@ibgreen @shaojingli , is there a better way instead of exporting it, so I can unit test this method?